app/models/candidate.rb

```diff
@@ -22,6 +22,12 @@ def self.for_email(email)
     find_or_initialize_by(email_address: email.downcase) if email
   end
 
+  def self.clear_application_data!(provider)
+    scope = joins(application_forms: { application_choices: { course_option: :course } })
+    scope.where("courses.accredited_provider": provider).or(scope.where("courses.provider": provider))
+      .delete_all
+  end
+
```

Comment:
Could we move this out to a small service?

Reply:
done